### PR TITLE
fix out of bounds texel fetch issues

### LINKED ISSUE #2257: dark rift background issue
GLideN64_DARK_RIFT_004

### PR DESCRIPTION
**Pull request by standard-two-simplex**
Fixes #2257

### CODE REVIEW

**Comment by tim241** [comment]
This seems to break texture packs somehow
This only breaks when you remove the `shaders` directory before changing builds
[build at commit](https://ci.appveyor.com/project/gonetz/gliden64/builds/32798994/artifacts): https://github.com/gonetz/GLideN64/commit/2616ba9587d8f42d53bbffbb7ddb29802be17c94
image
image

[build at commit](https://ci.appveyor.com/project/gonetz/gliden64/builds/32798994/artifacts):
https://github.com/standard-two-simplex/GLideN64/commit/631b4dc70b7b3b80aec4a154ca06c633e4788722
image
image

it works fine without the texture pack:
image

**Comment by standard-two-simplex** [comment]
Ok, thanks. I will check tomorrow.

**Comment by gonetz** [comment]
Fix:
```
			aTexSize[t][0] = pTexture->width * pTexture->hdRatioS;
			aTexSize[t][1] = pTexture->height * pTexture->hdRatioT;

```

File: src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramUniformFactory.cpp [hunk 1]
```diff
@@ -974,6 +980,9 @@ class UTextureEngine : public UniformGroup
 			if (pTile == nullptr || pTexture == nullptr)
 				continue;
 			
+			aTexSize[t][0] = pTexture->width;
+			aTexSize[t][1] = pTexture->height;
```
Comment (gonetz) [inline src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramUniformFactory.cpp:984]: aTexSize[t][0] = pTexture->width * pTexture->hdRatioS;
			aTexSize[t][1] = pTexture->height * pTexture->hdRatioT;

**Comment by gonetz** [comment]
Merged: f5ee3523e

**Pull request closed without merging**